### ISSUE TITLE
Hexo corrupting file when generating

### ISSUE THREAD

**Issue by justincalleja** [hexojs/hexo#1410]
Hi.

I have a minified js file which you can get from [http://pastebin.com/jjSXwxJR](http://pastebin.com/jjSXwxJR) or from checking out this repo and building with `grunt build`: [https://github.com/justin-calleja/compositeview-eg](https://github.com/justin-calleja/compositeview-eg).

I have settled on putting the file in `source/directoryForPost/js` as a new post is not created with the contents of the JS file if I put the post there (unlike when I put it in `source/_posts/directoryForPost/js`).

I thought that had solved the problem and I believe this was working for a while...

Now (after deploying) I am getting syntax errors in this file only when running in Hexo... If I diff the original file with the one in public after `hexo generate` I get this: [https://www.diffchecker.com/swhwbdlx](https://www.diffchecker.com/swhwbdlx)

It seems like Hexo is not simply copying the file over but changing it...

.... OK so I just remembered I had removed this:

```yaml
skip_render:
 - '**/*.js'
```

from the _config.yml because since it wasn't preventing the new post from being generated when JS file was in  `source/_posts/directoryForPost/js`, I figured it wasn't working so I removed it.

It seems like this is the trick to avoid Hexo from corrupting the JS file when moving over... honestly this seems a bit messy.

Is there anywhere I can put a file in Hexo so it just copies them over to `public` without having to guard against it corrupting the file?

PS I am keeping the "train of thought" above even though I managed a "work around" in order to show the confusion that an inexperienced Hexo user can come up against. I guess that is the issue I'm submitting here. Could this be made better or documented better?

PPS the diff and paste bin expire after a month.

Many thanks

**Comment by leesei** [comment]
I think you actually need this in order **not** to modify the js files

```
skip_render:
 - **/*.js
```

**Comment by justin-calleja** [comment]
Hi @leesei. Yes I ended up putting that bit of config back in and it's not modifying the js file - so that's good.

What I was finding confusing at the time was that, although I had "skip_render" of *.js files, the JS file for a particular post (in `source/_posts/directoryForPost/js`) was being rendered as a HTML page (a separate blog post made up of the JS source code...).

I thought "skip_render" would avoid that, but it didn't, so I removed "skip_render". Later I found out that if I put the JS file in `source/directoryForPost/js` as opposed to `source/_posts/directoryForPost/js`, *no new blog post for the JS file is created* - just what I wanted.

After deploying, I realised that the JS file wasn't working. After some debugging, I find that Hexo was changing the JS file *not just copying it*.

So basically, in order to copy a JS file over unchanged, you need to both have the file outside `source/_posts` e.g. `source/directoryForPost/js` and set "skip_render".

I figured this out while writing this issue and have submitted it anyway in case there's a better way someone can point me to or, if there isn't, it would make sense to point this out in Hexo documentation (or maybe it is and I skipped it... in that case feel free to close this issue).

**Comment by leesei** [comment]
Thanks for the testing result.
There's a specific option relating to post asset, I have yet to try it, I'm using a single asset folder in `source/`.
You can look into it and see if the behavior is different.
[Asset Folders | Hexo](https://hexo.io/docs/asset-folders.html)
